### DISCUSSION TITLE
Is there a way to filter by custom fields in graphql queries?

### DISCUSSION THREAD

**Discussion by emil-nasso** [netbox-community/netbox#7569] [Q&A]
I'm trying to filter tenants (in the `tenant_list`-query)  using a custom field.

In the REST-api, I can use this: https://netbox.readthedocs.io/en/stable/rest-api/filtering/#filtering-by-custom-field

I can't seem to find anything similar in the graphql-api.

The `tenant_list` query has a `q: String` argument though, not sure if I can input a custom filter there, somehow.

In this comment, there is mention that a solution is still needed for custom fields. https://github.com/netbox-community/netbox/issues/2007#issuecomment-863500573

Maybe custom fields are simply not supported (yet?) when filtering?

**Comment by candlerb** [comment]
I think you're right.  Although the [documentation](https://netbox.readthedocs.io/en/stable/graphql-api/overview/#filtering) says *"The GraphQL API employs the same filtering logic as the UI and REST API"*, filters like `cf_foo:"bar"` are rejected as unknown.

Probably worth raising as a feature request, or (if rejected) as a documentation update.

**Reply by emil-nasso** [reply]
I'll do that! :+1:

**Reply by emil-nasso** [reply]
Issue: https://github.com/netbox-community/netbox/issues/7598
